### PR TITLE
[merged] Improve error messages system containers

### PR DESCRIPTION
**Pull request by giuseppe**
Fail early if runc is not installed when installing a system container, or either if bwrap-oci is not installed or systemctl doesn't support --user when installing an user container.

### CODE REVIEW

File: Atomic/syscontainers.py [hunk 1]
```diff
@@ -141,6 +142,18 @@ def pull_image(self):
...
 
     def install_user_container(self, image, name):
+        bwrap_oci_found = True
+        try:
+            util.check_call([BWRAP_OCI_PATH, "--version"], stdout=DEVNULL)
+        except util.FileNotFound:
+            bwrap_oci_found = False
+
+        if not bwrap_oci_found:
+            raise ValueError("Cannot install the container: bwrap-oci is needed to run user containers")
```
Comment (rhatdan) [inline Atomic/syscontainers.py]: Why node raise this in the exception?

File: Atomic/syscontainers.py [hunk 2]
```diff
@@ -141,6 +142,18 @@ def pull_image(self):
...
+
+        if not "--user" in str(util.check_output(["systemctl", "--help"], stdin=DEVNULL, stderr=DEVNULL)):
+            raise ValueError("Cannot install the container: systemctl does not support --user")
+
```
Comment (rhatdan) [inline Atomic/syscontainers.py]: I think this checking should be done in the argparser.

File: Atomic/syscontainers.py [hunk 3]
```diff
@@ -149,6 +162,16 @@ def install_system_container(self, image, name):
...
+        if not self.user:
+            runc_found = True
+        try:
+            util.check_call([RUNC_PATH, "--version"], stdout=DEVNULL)
+        except util.FileNotFound:
+            runc_found = False
+
+        if not runc_found:
+            raise ValueError("Cannot install the container: runc is needed to run system containers")
+
```
Comment (rhatdan) [inline Atomic/syscontainers.py]: Why not raise this in the exception.  Don't see the need for runc_found.

Comment (giuseppe) [inline Atomic/syscontainers.py]: Python3 complains when you raise an exception while handling another one

File: Atomic/syscontainers.py (hunk 2 above)
Comment (giuseppe) [inline Atomic/syscontainers.py]: the advantage of doing it here is that Atomic.syscontainers can be used as a library from other applications and this check would still be done

File: Atomic/syscontainers.py (hunk 3 above)
Comment (rhatdan) [inline Atomic/syscontainers.py]: Does the following work?
```
 except util.FileNotFound as e:
         raise ValueError("Cannot install the container: runc is needed to run system containers") from e
```

Comment (giuseppe) [inline Atomic/syscontainers.py]: thanks, I have tested that if the exception is caught then Python3 doesn't error out, I have updated and rebased the patches.

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**